### ISSUE TITLE
Spyder issue I

### ISSUE THREAD

**Issue by PencilGraphite** [spyder-ide/spyder#17769]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x ] Tried basic troubleshooting (if a bug/error)
    * [ x] Restarted Spyder
    * [x ] Reset preferences with ``spyder --reset``
    * [x ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x ] Tried the other applicable steps from the Troubleshooting Guide
* [ x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I have been using Spyder 4.2 for roughly a year without issue. Upon opening and coding for roughly 15 minutes today the Application started freezing mid program constantly. Restarting the kernal unfreezes it, then upon running code same issue. Starts freezing. I uninstalled and reinstalled version. 5.1.5 . Same issue. I get not Tracebacks upon hitting the stop button when it freezes and no internal error messages now. Although I did get some internal messages on version 4.2 before uninstalling.

Was freezing during these lines. 
```
    display_hand(hand)
    print("do not break python")
    word = input("Enter a word from your hand: ")
    print("here")
    print("Or Enter !! if you would like to end your hand")
    while word != '!!':         #checking for end turn characters
        while n != 0:              #making sure hand has letters present
            if is_valid_word(word, hand, word_list) == True:
                print(get_word_score(word, n)) #displaying score
```

While executing the below function. Now it just freezes on pressing run.

```
def play_hand(hand, word_list)
    total = 0
    n = HAND_SIZE
    display_hand(hand)
    print("do not break python")
    word = input("Enter a word from your hand: ")
    print("here")
    print("Or Enter !! if you would like to end your hand")
    while word != '!!':         #checking for end turn characters
        while n != 0:              #making sure hand has letters present
            if is_valid_word(word, hand, word_list) == True:
                print(get_word_score(word, n)) #displaying score
                total += get_word_score(word, n) #running total of score from hand
                print(total) #displaying total
                hand = update_hand(hand, word) #removing used letters and setting variable appropriately
                n = calculate_handlen(hand) #counting letters in hand used for scoring
                display_hand(hand)
            else:
                print('This is not a valid word')
                hand = update_hand(hand, word)
                n = calculate_handlen(hand)#counting letters in hand for scoring
                display_hand(hand)
    return total
```

Any help on this would be greatly appreciated!

### What steps reproduce the problem?

1. Unsure
2. 
3. 

### What is the expected output? What do you see instead?
Expected output is at least a traceback where I made a mistake etc. Instead of an all out freezing.

### Paste Traceback/Error Below (if applicable)
```
ite-packages\spyder\plugins\ipythonconsole\widgets\debugging.py", line 278, in _handle_input_request
return super(DebuggingWidget, self)._handle_input_request(msg)
File “C:\Users\xxxxx\anaconda3\lib\site-packages\qtconsole\frontend_widget.py”, line 512, in _handle_input_request
self._readline(msg[‘content’][‘prompt’], callback=callback, password=msg[‘content’][‘password’])
File “C:\Users\xxxxx\anaconda3\lib\site-packages\qtconsole\console_widget.py”, line 2422, in _readline
self._show_prompt(prompt, newline=False, separator=False)
TypeError: _show_prompt() got an unexpected keyword argument ‘separator’
```

PASTE TRACEBACK HERE

## Versions

* Spyder version: * Spyder version: 5.1.5 None
* Python version: 3.9.7 64-bit
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  8.2.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jsonschema >=3.2.0            :  4.4.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.2 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.3.0 (OK)
qtpy >=1.5.0                  :  2.0.1 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  61.2.0 (OK)
sphinx >=0.6.6                :  4.4.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  None (OK)
numpy >=1.7                   :  None (OK)
pandas >=1.1.1                :  None (OK)
scipy >=0.17.0                :  None (OK)
sympy >=0.7.3                 :  None (OK)

```
PASTE DEPENDENCIES HERE
```

**Comment by PencilGraphite** [comment]
Okay so the issue has been improved following instructions from #17616 but I still can't run in my base environment. I think I understand why since my base environment is trying to run Python 3.9.7 64-bit, instead of 3.10.xx.

What is the best way of going about updating Python in my base environment from the Conda Forge environment?

Also, why would version 4.2 suddenly suffer from these symptoms(input internal error issue) as well as 5.1.5 if I were to want to run an older version? I'm trying to understand why these issues cropped up when I had ran fine for a year and had not updated/changed anything on my own?

Upon updating/uninstalling/reinstalling etc etc I feel like I'm getting an abundance of obsolete files in my base environment even though I have used clean uninstall when doing these things so if there are any comments on that I would be interested to know also. 

Thanks again for any and all advice or information relating to this!!!

**Comment by PencilGraphite** [comment]
Yeah, I will probably need some more assistance. I tried doing what I thought would work and now I can't even open Spyder in the anaconda prompt or in any form that I am aware of. So, I am transferring everything back to how I had it when I last commented. I appreciate the help

**Comment by PencilGraphite** [comment]
By additional help I mean, structuring my base environment to be used as default PATH for Spyder if that is possible with the current status of my issue.

**Comment by PencilGraphite** [comment]
Okay, so upon doing some more work in Spyder since I started using the conda-forge environment I'm noticing some weird interactions like print statements on separate lines showing up as one line. As well as a freeze on opening of Spyder at input lines. Clicking stop. Running the program again and it goes through the code fine except for the double print issue. I know it will be hard for you to tell if the output is an error in the code or not without posting the whole code since it's made up of functions but I'm double checking it to the best of my ability and it shouldn't be. If you would like me to post the whole code, I will so that you may verify. 


Output:
Loading word list from file...
   83667 words loaded.
Welcome! This game is played similar to scrabble
You will make words with the letters provided to you at beginning of each hand
Wildcards are represented by the * symbol, They may be played in place of a vowel only
Enter number of hands you would like to play: 2
* o a c m h p 
Would you like to substitute any letters in your hand?: no
* o a c m h p Enter a word from your hand: map                           ####Sometimes this prints on the same line. Other times it 
                                                                                                                doesn't (function call and input)
                                                             

Or Enter !! if you would like to end your handEnter a word from your hand:       #same issue here (print and input)
63
Total Score: 63
* o c h 



Code posted below:
```
def play_hand(hand, word_list):
    total = 0
    n = HAND_SIZE                                                          
    display_hand(hand)                                                   ######Lines printing on the same line
    word = input("Enter a word from your hand: ")        ######
    print("Or Enter !! if you would like to end your hand") #####
    while n != 0:        #checking for letter in hand                        
        while word != '!!':              #end turn character
            if is_valid_word(word, hand, word_list) == True:
                print(get_word_score(word, n)) #displaying score
                total += get_word_score(word, n) #running total of score from hand
                print("Total Score:", total) #displaying total
                hand = update_hand(hand, word) #removing used letters and setting variable appropriately
                n = calculate_handlen(hand) #counting letters in hand used for scoring
                display_hand(hand)
                word =  input("Enter a word from your hand:")
            else:
                print('This is not a valid word')
                hand = update_hand(hand, word)
                n = calculate_handlen(hand)#counting letters in hand for scoring
                display_hand(hand)
                word = input("Enter a word from your hand:")
    return total      
    ```
Display_hand function below
```
def display_hand(hand):
    for letter in hand.keys():
        for j in range(hand[letter]):  
             print(letter, end=' ')      # print all letters in hand on the same line
    print()      
```

```
 while num_hands >= 0:
        hand = deal_hand(HAND_SIZE)
        display_hand(hand)                                                                                                 ####These lines are the same issue
        sub_answer = input('Would you like to substitute any letters in your hand?: ')     ####
```

**Comment by ccordoba12** [comment]
> Okay, so upon doing some more work in Spyder since I started using the conda-forge environment I'm noticing some weird interactions like print statements on separate lines showing up as one line

That's a duplicate of issue #17710. We'll try to fix it in a future version.

**Comment by PencilGraphite** [comment]
I know this may be the best work around possible at the moment, but my Spyder is nearly unusable. With the jumbling/mixing of input/print statements it's making testing very difficult. Since every other time of running a program it's printing out and taking in inputs differently. Are there any other options such as Spyder being installed as a standalone without Anaconda that would work to help with this? I am willing to delete Anaconda until I am able to update appropriately through the Anaconda interface if necessary.

**Comment by ccordoba12** [comment]
> Are there any other options such as Spyder being installed as a standalone without Anaconda that would work to help with this?

No, there aren't, sorry. This bug was introduced when trying to fix another bug reported for years about the `input` statement introducing spurious blanks lines.

**Comment by CAM-Gerlach** [comment]
As mentioned on the other issue, if you are running Spyder from a Conda environment, you can temporarily downgrade QtConsole to 5.2.x until the bug is solved. With the environment in which you have installed Spyder activated, you could run e.g. the following:

```shell
conda install -c conda-forge qtconsole=5.2
```
